libs/coin-modules/coin-cosmos/src/bridge/js.ts

```diff
@@ -0,0 +1,128 @@
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+import { getCryptoCurrencyById } from "@ledgerhq/cryptoassets";
+import type { AccountBridge, CurrencyBridge } from "@ledgerhq/types-live";
+import type {
+  CosmosAccount,
+  CosmosCurrencyConfig,
+  CosmosValidatorItem,
+  Transaction,
+  TransactionStatus,
+} from "../types";
+import { asSafeCosmosPreloadData, setCosmosPreloadData } from "../preloadedData";
+import { CosmosValidatorsManager } from "../CosmosValidatorsManager";
+import { estimateMaxSpendable } from "../estimateMaxSpendable";
+import getTransactionStatus from "../getTransactionStatus";
+import { prepareTransaction } from "../prepareTransaction";
+import { updateTransaction } from "../updateTransaction";
+import { createTransaction } from "../createTransaction";
+import { getAccountShape } from "../synchronisation";
+import { buildSignOperation } from "../signOperation";
+import cryptoFactory from "../chain/chain";
+import { CosmosAPI } from "../api/Cosmos";
+import {
+  assignFromAccountRaw,
+  assignToAccountRaw,
+  fromOperationExtraRaw,
+  toOperationExtraRaw,
+} from "../serialization";
+import {
+  makeAccountBridgeReceive,
+  makeScanAccounts,
+  makeSync,
+} from "@ledgerhq/coin-framework/bridge/jsHelpers";
+import { CoinConfig, getCoinConfig, setCoinConfig } from "../config";
+import getAddressWrapper from "@ledgerhq/coin-framework/bridge/getAddressWrapper";
+import { SignerContext } from "@ledgerhq/coin-framework/signer";
+import resolver from "../hw-getAddress";
+import { CosmosSigner } from "../types/signer";
+
+const sync = makeSync({ getAccountShape });
+
+function buildCurrencyBridge(signerContext: SignerContext<CosmosSigner>): CurrencyBridge {
+  const getAddress = resolver(signerContext);
+
+  const scanAccounts = makeScanAccounts({
+    getAccountShape,
+    getAddressFn: getAddressWrapper(getAddress),
+  });
+
+  const getPreloadStrategy = () => ({
+    preloadMaxAge: 30 * 1000,
+  });
+
+  return {
+    getPreloadStrategy,
+    preload: async (currency: CryptoCurrency) => {
```

Comment:
Extract `getPreladStrategy`, `preload` and `hydrate into a separate file (like `preload.ts`).
You can see in `coin-tron` as an example.

Reply:
done in https://github.com/LedgerHQ/ledger-live/pull/7636/commits/bab4e8a72e85add3873868d734d606b44e33b993